optimade/server/routers/utils.py

```diff
@@ -285,6 +286,10 @@ def retrieve_queryable_properties(schema: dict, queryable_properties: list) -> d
                 # All properties are sortable with the MongoDB backend.
                 # While the result for sorting lists may not be as expected, they are still sorted.
                 properties[name]["sortable"] = True
+                # Try to get OpenAPI-specific "format" if possible, else get "type"; a mandatory OpenAPI key.
+                properties[name]["type"] = DataType.from_json_type(
+                    value.get("format", value["type"])
+                )
```

Comment:
Could you explain the comment here? Is the format not OPTIMADE-specific?

Reply:
No, What we're dealing with here is the `schema()` output from a `pydantic` model, i.e., it's the OpenAPI schema. So we're dealing with the data types specified in the OpenAPI specification. A nice overview is given [here](https://swagger.io/docs/specification/data-models/data-types/).
They are also basing it on JSON Schema, but they're diverging from it, e.g., adding the `"format"` property/field as well as making `"type"` mandatory.

Reply:
Gotcha!